contracts/EthernalGatesReview.sol

```diff
@@ -0,0 +1,164 @@
+pragma solidity 0.8.11;
+
+/// @author Hammad Ghazi
+contract EthernalGatesReview is ERC721A, Ownable {
+    using MerkleProof for bytes32[];
+
+    enum SALE_STATUS {
+        OFF,
+        INVESTOR,
+        VIP,
+        WL,
+        PUBLIC
+    }
+
+    SALE_STATUS public saleStatus;
+
+    string baseTokenURI;
+
+    // Max Supply of Ethernal Gates
+    uint256 public constant MAX_SUPPLY = 6000;
+
+    // Holds max number of NFTs that can be minted at the moment
+    uint256 public currentSupply = 2000;
+
+    uint256 public presalePrice = 0.49 ether;
+    uint256 public publicPrice = 0.59 ether;
+
+    bytes32 public merkleRoot;
+
+    // To store NFTs a particular address has minted in each whitelist phase
+    mapping(address => uint256) public investorMintCount;
+    mapping(address => uint256) public vipMintCount;
+    mapping(address => uint256) public wlMintCount;
+
+    constructor(string memory baseURI) ERC721A("Ethernal Gates", "Ethernal Gates") {
+        setBaseURI(baseURI);
+    }
+
+    modifier soldOut(uint256 _count) {
+        require(
+            totalSupply() + _count <= currentSupply,
+            "Transaction will exceed maximum available supply of Ethernal Gates"
+        );
+        _;
+    }
+
+    // Admin only functions
+
+    // To update sale status
+    function setSaleStatus(SALE_STATUS _status) external onlyOwner {
+        saleStatus = _status;
+    }
+
+    function setMerkleRoot(bytes32 _merkleRoot) external onlyOwner {
+        merkleRoot = _merkleRoot;
+    }
+
+    function changePresalePrice(uint256 _presalePrice) external onlyOwner {
+        presalePrice = _presalePrice;
+    }
+
+    function changePublicPrice(uint256 _publicPrice) external onlyOwner {
+        publicPrice = _publicPrice;
+    }
+
+    // To increase the supply, can't exceed 6000
+    function increaseSupply(uint256 _increaseBy) external onlyOwner {
+        require(currentSupply + _increaseBy <= MAX_SUPPLY, "Cannot increase supply by more than 6000");
+        currentSupply += _increaseBy;
+    }
+
+    function withdraw() external onlyOwner {
+        (bool success, ) = payable(msg.sender).call{value: address(this).balance}("");
```

Comment:
♻️The .call` method works on the `address` type; there's no need to typecast to `address payable`.
📝 Good approach for transferring ETH using `call` instead of `send` or `transfer`. We are in an `onlyOwner` function, and there are no concerns about reentrancy.
```suggestion
        (bool success, ) = msg.sender.call{value: address(this).balance}("");
```

Reply:
done